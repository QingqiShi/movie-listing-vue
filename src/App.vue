<template>
  <div id="app" class="app--mounted">
    <template v-if="loading">
      <div class="app__load-wrapper">
        <spinner />
      </div>
    </template>
    <template v-else>
      <router-view />
    </template>
  </div>
</template>

<script>
import Spinner from "@/components/Spinner.vue";

export default {
  components: {
    Spinner
  },
  data: () => ({
    loading: true
  }),
  async mounted() {
    await this.$store.dispatch("initialFetch");
    this.loading = false;
  }
};
</script>

<style lang="scss">
@import "@/styles/_reset.scss";

.app {
  &__load-wrapper {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    display: flex;
    justify-content: center;
    align-items: center;
  }
}
</style>
