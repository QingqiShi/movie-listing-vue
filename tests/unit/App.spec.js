import { shallowMount, createLocalVue } from "@vue/test-utils";
import Vuex from "vuex";
import App from "@/App.vue";

const localVue = createLocalVue();
localVue.use(Vuex);

describe("App.vue", () => {
  describe("update sort", () => {
    jest.useFakeTimers();
    let wrapper;

    beforeAll(() => {
      const store = new Vuex.Store({
        actions: {
          initialFetch: () => new Promise(resolve => setTimeout(resolve, 100))
        }
      });
      wrapper = shallowMount(App, {
        store,
        localVue,
        stubs: { "router-view": true }
      });
    });

    it("renders", async () => {
      expect(wrapper).toMatchSnapshot();

      jest.runAllTimers();
      // Run promise callback
      await wrapper.vm.$nextTick();
      // Component render
      await wrapper.vm.$nextTick();

      expect(wrapper.vm.loading).toBe(false);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
