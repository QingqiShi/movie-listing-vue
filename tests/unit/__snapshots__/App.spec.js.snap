// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue update sort renders 1`] = `
<div id="app" class="app--mounted">
  <div class="app__load-wrapper">
    <spinner-stub></spinner-stub>
  </div>
</div>
`;

exports[`App.vue update sort renders 2`] = `
<div id="app" class="app--mounted">
  <router-view-stub></router-view-stub>
</div>
`;
